packages/purpose-process/src/services/purposeService.ts

```diff
@@ -550,6 +556,68 @@ export function purposeServiceBuilder(
         totalCount: purposesList.totalCount,
       };
     },
+    async createPurpose(
+      purposeSeed: ApiPurposeSeed,
+      organizationId: TenantId,
+      correlationId: string
+    ): Promise<{ purpose: Purpose; isRiskAnalysisValid: boolean }> {
+      logger.info(
+        `Creating Purpose for EService ${purposeSeed.eserviceId} and Consumer ${purposeSeed.consumerId}`
+      );
+      const eserviceId = unsafeBrandId<EServiceId>(purposeSeed.eserviceId);
+      const consumerId = unsafeBrandId<TenantId>(purposeSeed.consumerId);
+      assertOrganizationIsAConsumer(organizationId, consumerId);
+
+      assertConsistentFreeOfCharge(
+        purposeSeed.isFreeOfCharge,
+        purposeSeed.freeOfChargeReason
+      );
+
+      const tenant = await retrieveTenant(organizationId, readModelService);
+
+      assertTenantKindExists(tenant);
+
+      const validatedFormSeed = validateAndTransformRiskAnalysis(
+        purposeSeed.riskAnalysisForm,
+        tenant.kind
+      );
+
+      const agreement = await readModelService.getActiveAgreement(
+        eserviceId,
+        consumerId
+      );
+
+      if (agreement === undefined) {
+        throw agreementNotFound(eserviceId, consumerId);
+      }
```

Comment:
in the `readModelService` we never throw specific errors, but only generic errors. for this reason in my opinion it would be more correct to leave it within the `purposeService`